local/sdk-sriov/pkg/k8s/deviceplugin/server_test.go

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deviceplugin
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+	podresources "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+
+	testingtools "github.com/networkservicemesh/cmd-forwarder-sriov/test/tools"
+)
+
+const (
+	resourceName    = "resource"
+	resourceCount   = 10
+	hostBaseDir     = "/base/dir"
+	hostPathEnv     = "HOST_PATH"
+	socket          = "socket"
+	containersCount = 5
+)
+
+var config = &ServerConfig{
+	ResourceName:  resourceName,
+	ResourceCount: resourceCount,
+	HostBaseDir:   hostBaseDir,
+	HostPathEnv:   hostPathEnv,
+}
+
+func TestDevicePluginServer_Start(t *testing.T) {
+	m := &managerMock{}
+	m.mock.On("StartDeviceServer", mock.Anything, mock.Anything).
+		Return(socket, nil)
+	m.mock.On("RegisterDeviceServer", mock.Anything, mock.Anything).
+		Return(nil)
+
+	resetCh := make(chan bool)
+	m.mock.On("MonitorKubeletRestart", mock.Anything).
+		Return(resetCh, nil)
+
+	m.mock.On("GetPodResourcesListerClient", mock.Anything).
+		Return(struct {
+			podresources.PodResourcesListerClient
+		}{}, nil)
+
+	err := StartServer(context.TODO(), config, m)
+	assert.Nil(t, err)
+
+	m.mock.AssertCalled(t, "StartDeviceServer", mock.Anything, mock.Anything)
+	m.mock.AssertCalled(t, "RegisterDeviceServer", mock.Anything, mock.MatchedBy(func(request *pluginapi.RegisterRequest) bool {
+		return request.Endpoint == socket && request.ResourceName == resourceNamePrefix+resourceName
+	}))
+
+	testingtools.WriteBoolChan(t, resetCh, true, 10*time.Second)
+	assertNumberOfCallsEventually(t, m, "RegisterDeviceServer", 2)
+}
+
+func assertNumberOfCallsEventually(t *testing.T, m *managerMock, methodName string, expectedCalls int) {
+	assert.Eventually(t, func() bool {
+		var count int
+		for i := range m.mock.Calls {
+			if m.mock.Calls[i].Method == methodName {
+				count++
+			}
+		}
+		return count == expectedCalls
+	}, 10*time.Second, 10*time.Millisecond)
+}
```

Comment:
```suggestion
	assert.Eventually(t, func() bool {
		var count int
		for i := range m.mock.Calls {
			if m.mock.Calls[i].Method == methodName {
				count++
			}
		}
		return count == expectedCalls
	}, 10*time.Second, 10*time.Millisecond)
}
```

Reply:
I do not agree that inlining functions with meaningful names increases code readability.

Reply:
+1 to use inline assert.

Reply:
Inlined